_posts/2018-07-17-its-a-lot-more-snappy.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: post
+title: "It's A Lot More Snappy"
+date: 2018-07-17
+categories: PWA SPA JavaScript
+author: Rick Foxcroft
+image: img/repeatable-serverless-applications/less_servers.png
+---
+
+**It's A Lot More Snappy**
+_‘A real world application of PWA technology’_
+
+Recently we undertook the journey of building a PWA(progressive web app) for a client here in Perth.  The app, is a maintenance app that allows maintenance technicians to raise issues with equipment in geologically  isolated areas. Due to this isolation, it was important that app could function in an offline mode and to do so, the app needed to store some data on the device it was running on.
+
+Whilst building the app, we encountered some performance issues. Over time the app would gradually become unresponsive to the point where, it would take a few seconds to register a button click.
+
+## What Was The Problem?
+
+* The app had to deal with a somewhat large dataset. One of these datasets was 205509 objects to be exact. Let’s refer to these objects as Pokemon.
+
+* The list of Pokemon had to be used in several places throughout the app and in some places not all of the data pertaining to each Pokemon was used.
+
+* A JSON file of this magnitude takes a fair amount of time to download.
```

Comment:
A JSON file of what magnitude? Is the dataset in the JSON file? Be clear. What is a "fair amount of time"? Suggest talking about the types of connections on the remote sites (eg, bandwidth) the types of devices used and their memory consumption on the JSON dataset being loaded in, and use some numbers if you have them or some generalisations on typical numbers. Eg:
Sites had a bandwidth capacity of 300kb/sec and for a JSON file which had our dataset in, this would take more than 3 minutes to download .... or something. Using numbers helps to paint a picture of the problem and then people can understand what you mean by magnitude